### ISSUE TITLE
Slug conflict when syncing forked docs

### ISSUE THREAD

**Issue by taycaldwell** [readmeio/rdme#519]
I have two Github Action workflows. Each one runs a Readme Doc Sync for specific versions.

**Sync v1 of docs**

```
name: GitHub Actions Demo
on: [push]
jobs:
  sync:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout this repo
        uses: actions/checkout@v3

      - name: GitHub Action
        uses: readmeio/rdme@7.1.0
        with:
          rdme: docs ./docs/v1 --key=<KEY> --version=1.0.0
```

**Sync v2 of docs**

```
name: GitHub Actions Demo
on: [push]
jobs:
  sync:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout this repo
        uses: actions/checkout@v3

      - name: GitHub Action
        uses: readmeio/rdme@7.1.0
        with:
          rdme: docs ./docs/v2 --key=<KEY> --version=2.0.0
```

I created v2 as a fork of v1 under `Manage versions` of the Readme UI.

The sync for v1 works fine, however the sync for v2 fails because of conflicting slugs:

```
Error uploading docs/v2/page.md: We couldn't save this doc (Page slug must be unique.).
```

Is this intended behavior? Unique slugs makes sense for a specific version, however, since new versions are forked from older versions, isn't it assumed that the slugs will be duplicated? It seems to be true when I query for slugs using the ReadMe API.

**Comment by kanadgupta** [comment]
Hi @taycaldwell!

> Unique slugs makes sense for a specific version, however, since new versions are forked from older versions, isn't it assumed that the slugs will be duplicated? It seems to be true when I query for slugs using the ReadMe API.

Slugs must be unique within a single version (i.e. the slug `example-slug` can only exist once within a single project version, e.g. `v1`) but can be the same across versions (i.e. the slug `example-slug` can exist within two different project versions, e.g. `v1` and `v2`).

I'm guessing this is either an issue with either your `slug` or `category` parameters. Could you verify the following:
- If your file names contain special characters, you've specified the slug via the front matter (more on this in [our docs](https://docs.readme.com/docs/rdme#specifying-page-slugs))
- The `category` parameter specified in your front matter belongs to the project version you're pointed to. You can get a list of all categories for a version using [this endpoint](https://docs.readme.com/reference/getcategories)[^1]

If you're still stuck on this, can you reach out to support@readme.io with your debug logs? There's info on how to do that [here](https://docs.readme.com/docs/rdme#troubleshooting).

[^1]: If this was in fact the problem, #497 is tracking our progress on fixing that confusing error message.

**Comment by taycaldwell** [comment]
=The problem was the ID for `category` changes when forking a new version. The error was confusing/misleading, as it mentioned there was a problem with the `slug`.

After updating the `category` for each doc, the sync now works as expected.

Thank you for the clarification and help!

**Comment by kanadgupta** [comment]
Agreed on the confusing error message, apologies for that. As mentioned above, we're tracking that fix in #497. Happy to help @taycaldwell!
